### PR TITLE
Implement `rad-push`

### PR DESCRIPTION
**Pull request by cloudhead**
Signed-off-by: Alexis Sellier <alexis@radicle.xyz>

### CODE REVIEW

File: radicle-cli/src/commands/push.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut verbose = false;
+        let mut force = false;
+        let mut all = false;
+        let mut sync = false;
+        let mut set_upstream = false;
```
Comment (slack-coder) [inline radicle-cli/src/commands/push.rs:55]: small suggestion, alphabetical order?

File: radicle-cli/src/commands/push.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    rad push [--all] [--[no-]sync] [<option>...]
+
+    By default, only the current branch is synced.
+
+Options
+
+    --all               Push all branches (default: false)
+    --sync              Sync after pushing to the "rad" remote (default: false)
+    --no-sync           Do not sync after pushing to the "rad" remote
```
Comment (FintanH) [inline radicle-cli/src/commands/push.rs:26]: Why have both `sync` and `no-sync` flags? :thinking: Seems weird to have and feels like it should have precedence rules if it they both stay, i.e. which one trumps the other?

Comment (cloudhead) [reply radicle-cli/src/commands/push.rs:26]: Good question - I usually have a `--no-*` version for many boolean flags as I think it's useful as self-documentation when you see the command written and to not force users to remember what the default is.

As far as I can tell this is pretty standard, eg. see this screenshot from the ripgrep manual:
image

File: radicle-cli/src/commands/push.rs (hunk 1 above)
Comment (cloudhead) [reply radicle-cli/src/commands/push.rs:55]: I don't think it's worth it for a list of 6 items?

File: radicle-cli/src/commands/push.rs (hunk 2 above)
Comment (FintanH) [reply radicle-cli/src/commands/push.rs:26]: Ya, so maybe we should follow suit and make `--no-sync` override any use of `--sync`.

Comment (cloudhead) [reply radicle-cli/src/commands/push.rs:26]: https://github.com/radicle-dev/heartwood/issues/106

File: radicle-cli/src/commands/push.rs (hunk 1 above)
Comment (slack-coder) [reply radicle-cli/src/commands/push.rs:55]: Having clear ordering makes it easier to add items.  Newcomers have to figure out if there's a hidden order or not when adding fields.  Its stylistic and not a big deal, but I find it reduces a small amount of friction.

**Pull request merged**